src/raster/rasterband.rs

```diff
@@ -476,8 +475,9 @@ impl<'a> RasterBand<'a> {
     }
 
     /// Returns the pixel datatype of this band.
-    pub fn band_type(&self) -> GDALDataType::Type {
-        unsafe { gdal_sys::GDALGetRasterDataType(self.c_rasterband) }
+    pub fn band_type(&self) -> GdalDataType {
+        let ordinal = unsafe { gdal_sys::GDALGetRasterDataType(self.c_rasterband) };
+        ordinal.try_into().unwrap_or(GdalDataType::Unknown)
```

Comment:
Just looking at the code in the browser, can we call `GdalType::datatype()` here instead?

Reply:
@lnicola Not sure what you mean... `GdalType::datatype()` would be to get the datatype most closely associated with a static primitive. In this case, we're getting a `c_int` from C-GDAL and need to map that into the `GdalDataType` enum. But perhaps I'm missing what you're suggesting?

Reply:
Basically `ordinal.datatype()` instead of `ordinal.try_into().unwrap_or(GdalDataType::Unknown)`. Maybe it doesn't work for `c_int`, though?

Reply:
Yeh, it doesn't work. `GdalType` is a (wierd) type-level trait, with no `self` parameter:

https://github.com/georust/gdal/blob/029b95af44096d34b1ce3dbb80fe43698b16ef70/src/raster/types.rs#L330

In Scala I'd have called this an "evidence" trait. So `<u8>::datatype()` exists, but `3u8.datatype()` does not.